include/rocksdb/iostats_context.h

```diff
@@ -50,7 +50,11 @@ struct IOStatsContext {
   uint64_t cpu_read_nanos;
 };
 
-// Get Thread-local IOStatsContext object pointer
+// Get Thread-local IOStatsContext object pointer if supported.
```

Comment:
Is there a reason this method definition cannot be conditionally defined?  If NIOSTATS_CONTEXT is defined, this method does not exist, and attempts to use it will fail at compile time.

Reply:
I think the reason is we do not have to complicate the API with `ifdef`. Even if `NIOSTATS_CONTEXT` is defined, this method can still exist and return a pointer, as in the case of `PerfContext` (`get_perf_context()`). See the comment of `get_iostats_context()` and `get_perf_context()`. Keeping the definition will be safe in all combinations of thread-local support and `-DNIOSTATS_CONTEXT` / `-DNPERF_CONTEXT`.